python_modules/dagster/dagster/_daemon/daemon.py

```diff
@@ -269,11 +275,30 @@ def core_loop(
         workspace_process_context: IWorkspaceProcessContext,
         shutdown_event: Event,
     ) -> DaemonIterator:
-        yield from execute_sensor_iteration_loop(
-            workspace_process_context,
-            self._logger,
-            shutdown_event,
-        )
+        with ExitStack() as stack:
```

Comment:
Let's follow the pattern of initializing the `ExitStack` on initialization + cleaning up  in `__exit__`.